src/sentry/api/endpoints/organization_invite_request_details.py

```diff
@@ -1,14 +1,210 @@
 from __future__ import absolute_import
 
+from django.db.models import Q
+from django.conf import settings
+from rest_framework import serializers, status
+from rest_framework.response import Response
 
-from sentry.api.bases.organization import OrganizationEndpoint
+from sentry import roles, features
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize, OrganizationMemberWithTeamsSerializer
+from sentry.models import AuditLogEntryEvent, InviteStatus, OrganizationMember
+from sentry.signals import member_invited
+
+from .organization_member_details import get_allowed_roles
+from .organization_member_index import OrganizationMemberSerializer, save_team_assignments
+
+ERR_CANNOT_INVITE = "Your organization is not allowed to invite members."
+ERR_INSUFFICIENT_ROLE = "You do not have permission to invite that role."
+ERR_INSUFFICIENT_SCOPE = "You are missing the member:write and/or member:admin scope."
+ERR_JOIN_REQUESTS_DISABLED = "Your organization does not allow requests to join."
+
+
+class ApproveInviteRequestSerializer(serializers.Serializer):
+    approve = serializers.BooleanField(required=True, write_only=True)
+
+    def validate_approve(self, approve):
+        request = self.context["request"]
+        organization = self.context["organization"]
+        member = self.context["member"]
+        allowed_roles = self.context["allowed_roles"]
+
+        if not features.has("organizations:invite-members", organization, actor=request.user):
+            raise serializers.ValidationError(ERR_CANNOT_INVITE)
+
+        if (
+            organization.get_option("sentry:join_requests") is False
+            and member.invite_status == InviteStatus.REQUESTED_TO_JOIN.value
+        ):
+            raise serializers.ValidationError(ERR_JOIN_REQUESTS_DISABLED)
+
+        if not request.access.has_scope("member:admin") and not request.access.has_scope(
+            "member:write"
+        ):
+            raise serializers.ValidationError(ERR_INSUFFICIENT_SCOPE)
+
+        # members cannot invite roles higher than their own
+        if member.role not in {r.id for r in allowed_roles}:
```

Comment:
Why do you have to validate this here? Don't we already validate that elsewhere?